modules/custom/dkan_datastore/src/DataNodeLifeCycle.php

```diff
@@ -71,20 +71,30 @@ public function predelete() {
   /**
    * Private.
    */
-  private function isDatastorable() {
+  private function isDataStorable() {
     $metadata = $this->getMetaData();
     $data = $metadata->data;
 
-    if (!isset($data->downloadURL) && !isset($data->accessURL)) {
-      return FALSE;
+    // Project Open Data schema requires distributions to contain downloadURL
+    // or accessURL. In case of downloadURL, mediaType is required as well.
+    if (isset($data->downloadURL) && isset($data->mediaType)) {
+      $acceptableMediaTypes = [
+        'text/csv',
+        'text/tab-separated-values',
+      ];
+      return in_array($data->mediaType, $acceptableMediaTypes);
     }
-
-    if (!(isset($data->mediaType) && $data->mediaType == 'text/csv') &&
-      !(isset($data->format) && $data->format == 'csv')) {
-      return FALSE;
+    if (isset($data->accessURL) && isset($data->format)) {
+      $acceptableFormat = [
+        'csv',
+        'tsv',
+        'tab',
+        'txt',
+      ];
+      return in_array($data->format, $acceptableFormat);
```

Comment:
@thierrydallacroce will the logic break here if the mediaType is empty string for example? I believe in that case it will return fall even if the format is set correctly and it is one of the accepted formats.

Reply:
Agreed, will revisit. I forgot about the slight schema differences between Dkan2 and our other project, where there's a dependency that requires mediaType when downloadURL is present.